### DISCUSSION TITLE
Can we customize the relation Manager query?

### DISCUSSION THREAD

**Discussion by StaffCollab** [filamentphp/filament#6975] [Help] [answered]
I have a relations manager for charges on a payment resource.

But that relationship (many-to-many, and because the table is doing additional things there are some duplicated lines) returns duplicated results, so I wanted to add a ->unique('id') to the relationship, how can I achieve this with a relations manager?

Actually, I have a charges attribute that is doing exactly this and I'm using it throughout my application instead of the relationship itself
This is my setup
~~~
    public function allCharges(): BelongsToMany
    {
        return $this->belongsToMany(
            AccountAdvertisementCharges::class,
            'account_payment_charges',
            'payment',
            'charge'
        );
    }

     public function getChargesAttribute()
     {
         return $this->allCharges
         ->unique('id')
         ;
     }
~~~
But as I cannot use a attribute as a relationship in the relations manager, I need to scope it, is there a easy way?

**Comment by rmitesh** [comment]
You override the `getRelationship()` function of RelationManager class, and add your conditions
```php
public function getRelationship(): Relation | Builder
{
    return parent::getRelationship()->latest();
}
```

**Reply by danharrin** [reply, marked as answer]
`getTableQuery()` is probably better in this case

**Reply by rmitesh** [reply]
@danharrin `getTableQuery()` internally used `getRelationship()` to get query builder, right?
I see it's better to use `getTableQuery()`.

**Reply by Nikkitory** [reply]
For those landing here on google, after version 3, its this on the relationshipmanager: 
`public function table(Table $table): Table
{
    return $table
        ->modifyQueryUsing(fn (Builder $query) => $query->where('is_active', true))
        ->columns([`
